docs/naps/8-multiple-canvases.md

```diff
@@ -0,0 +1,261 @@
+
+[^napari-lite]: A lightweight "canvas" might be relevant to the implementation of ["napari-lite"](https://github.com/napari/napari/issues/5940).
+
+An important consideration is to minimize breaking changes to the public napari API. While napari is still pre-1.0, there is already a healthy developing ecosystem of plugins, scripts, and users. Changes to the API may be necessary and should be made if they constitute improvements, but should be minimized and well documented.
+
+The concept of an "active" canvas will work in service of minimizing API changes. This will allow existing APIs on the main Viewer/ViewerModel to remain and simply delegate to the active canvas.
+
+In addition to maintaining the MVC architecture of napari, this proposal aims to maintain or improve decoupling of the UI framework (currently Qt), the visualization library (currently VisPy), and the napari core code.
+
+> TODO: add a list open questions and key decisions here [name=Ashley A]
+> * selection state
+
+### Architecture
+napari architecture is based on the MVC pattern. The model layer comprises a `ViewerModel` and a list of Layer models (subclasses of a base `Layer`). There are seven layer types, each with a corresponding view type. Currently models and views are paired 1:1, and the correlation is stored in a map (`layer_to_visual`) on the `VispyCanvas`. Figure 1 shows the class relationships for the base model types and the Image layer types (for brevity - other layer types have similar connectivity).
+
+```{image} _static/multicanvas-napari-architecture-today.png
+---
+name: fig-1
+---
+Fig. 1: napari architecture today.
+```
+
+Figure 2 shows proposed changes (in orange) to the architecture to support multiple canvases. The new architecture is still following the MVC pattern. Again, this diagram only includes the Image layer type. Here is a summary of the planned changes:
+* Slice state will be moved off the layer as necessary, into new `LayerSlice` classes for each layer type
+    * This will be different for each Layer type - unifying the structure in the process may be a secondary benefit but is not the goal
+* Each VispyCanvas will hold a reference to a dedicated model class (`CanvasModel`)
+    * dims will move from `ViewerModel` -> `CanvasModel`
+    * camera will move from `ViewerModel` -> `CanvasModel`
+    * `layer_to_slice` will map each Layer (global list) to a `LayerSlice` (one per `CanvasModel`)
+* `ViewerModel` will own a *list* of `CanvasModel` objects
+* `QtViewer` will own a *list* of `VispyCanvas` objects
+* `VispyLayer` subclasses will hold references to their `Layer` (for rendering information) as well as a `LayerSlice` (for data)
+* The LayerSlicer will need to update the sync and async callbacks[^async-only] to pass a canvas parameter where the resulting sliced data will be stored, rather than storing it on the layer itself. Slice task cancellation logic will need to be revisited accordingly.
```

Comment:
Thanks - that's an interesting idea I hadn't considered. I was initially thinking the current LayerSlicer may expand to use a pool instead of a single thread.

I think you're right passing the full canvas is probably more than necessary, and trying to pass a canvas ID or something feels a bit hacky to me. One benefit of just passing the canvas object is that it would bring with it the camera, which might be useful for finding `corner_pixels` without waiting for the draw.